### PR TITLE
fix metrics kubelet_running_pod_count

### PR DESCRIPTION
**Pull request by xiaoanyunfei**
**What type of PR is this?**

/kind bug

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #https://github.com/kubernetes/kubernetes/issues/85450

**Special notes for your reviewer**:

I'm not sure `kubelet_running_pod_count` should include situation when sandbox is running and container is exited. I can fix it if necessary
 
**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```
/sig node

### CODE REVIEW

**Comment by mnaser** [comment]
This issue is affecting me and this seems like a solid fix.  The reasoning behind why we end up with a lot of pods that are not running is backups which spawn jobs that aren't cleaned up until much later.

I would appreciate if reviewers on this would take sometime to look into this :)

**Review by mnaser — APPROVED** [review]

**Comment by sjenning** [comment]
also lgtm.  I don't have approver here.
/cc @derekwaynecarr

**Comment by renan** [comment]
LGTM

It would be nice to have this released as soon as possible.

**Comment by johngmyers** [comment]
Confirming that between 1.15 and 1.16 the metric changed from a value obtained from `Runtime.GetPods(false)` to a value obtained from `Runtime.GetPods(true)`.

File: pkg/kubelet/pleg/generic.go [hunk 1]
```diff
@@ -445,10 +444,23 @@ func updateRunningPodAndContainerMetrics(pods []*kubecontainer.Pod) {
 			// update the corresponding "container_state" in map to set value for the gaugeVec metrics
 			containerStateCount[string(container.State)]++
 		}
+
+		sandboxes := pod.Sandboxes
```
Comment (derekwaynecarr) [inline pkg/kubelet/pleg/generic.go:448]: just noting how this regressed:

https://github.com/kubernetes/kubernetes/commit/c02d49d775b4dc960f52af1f5295642c07947ca7

moved a metric that was sourced from the kubelet runtime cache to now be sourced by the pleg directly.

**Review by derekwaynecarr — APPROVED** [review]
/approve
/lgtm

**Comment by derekwaynecarr** [comment]
Thank you for the fix.  Can you open cherrypicks?

**Comment by xiaoanyunfei** [comment]
> Thank you for the fix. Can you open cherrypicks?

What is cherrypicks?

**Comment by lilic** [comment]
Seems like we also have https://github.com/kubernetes/kubernetes/pull/92187 that tries to solve a similar problem.

**Pull request merged**